src/parquet/column/reader.h

```diff
@@ -184,7 +188,7 @@ inline int64_t TypedColumnReader<DType>::ReadBatch(int batch_size, int16_t* def_
   }
 
   // Not present for non-repeated fields
-  if (descr_->max_repetition_level() > 0) {
+  if (descr_->max_repetition_level() > 0 && rep_levels) {
     num_rep_levels = ReadRepetitionLevels(batch_size, rep_levels);
     if (num_def_levels != num_rep_levels) {
       throw ParquetException("Number of decoded rep / def levels did not match");
```

Comment:
I'm still not totally grasping this. Basically it sounds like Spark does not let users write non-nullable columns. So with the data being the data, the correct answer is to always pass a def levels buffer, even if it's all 1's. Ordinarily of `max_definition_level() > 0` then def_levels should never be done. It sounds like you are passing nullptr as a performance optimizing, is that right?